tests/test_sh_class.py

```diff
@@ -6,16 +6,101 @@
 import pyfar as pf
 from spharpy import SphericalHarmonics
 from spharpy.samplings import gaussian, calculate_sampling_weights, equiangular
+from spharpy.classes.sh import SphericalHarmonicDefinition
+
+
+def test_spherical_harmonics_definition_init():
+    """Test default behavior.
+    """
+    definition = SphericalHarmonicDefinition()
+    assert definition.basis_type == 'real'
+    assert definition.normalization == 'n3d'
+    assert definition.channel_convention == 'acn'
+
+
+@pytest.mark.parametrize("phase_convention", [True, False])
+def test_setter_phase_convention(phase_convention):
+    sph_harm = SphericalHarmonicDefinition()
+    sph_harm.condon_shortley = phase_convention
+    assert sph_harm.condon_shortley == phase_convention
+
+
+def test_init_phase_convention_auto():
+    sph_harm = SphericalHarmonicDefinition(
+        basis_type="complex",
+        condon_shortley="auto")
+    assert sph_harm.condon_shortley is True
+
+    sph_harm = SphericalHarmonicDefinition(
+        basis_type="real",
+        condon_shortley="auto")
+    assert sph_harm.condon_shortley is False
+
+def test_setter_phase_convention_auto():
+    sph_harm = SphericalHarmonicDefinition()
+    sph_harm.basis_type = "complex"
+    sph_harm.condon_shortley = "auto"
+    assert sph_harm.condon_shortley is True
+
+    sph_harm = SphericalHarmonicDefinition()
+    sph_harm.basis_type = "real"
+    sph_harm.condon_shortley = "auto"
+    assert sph_harm.condon_shortley is False
+
+
+def test_setter_phase_convention_invalid():
+    sph_harm = SphericalHarmonicDefinition()
+    with pytest.raises(TypeError):
```

Comment:
Add a `match=str` please. If this does not raise a ruff error, we could check for consistency of the ruff rules with respect to pyfar, where it would. @ahms5 - if I remember correctly from the weekly, you wanted to do this anyhow, right?